### PR TITLE
mlperf-deepcam: fixed issue with cuda variant and deps

### PR DESCRIPTION
**Pull request by alalazo**
Some directives were referring to a non-existing "cuda" variant. Some dependencies were referring non-existing variants in their metadata. Spotted using #23053

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-apache-beam/package.py [hunk 1]
```diff
@@ -18,7 +18,6 @@ class PyApacheBeam(PythonPackage):
     depends_on('py-setuptools', type='build')
     depends_on('py-pip@7.0.0:', type=('build', 'run'))
     depends_on('py-cython@0.28.1:', type=('build', 'run'))
-    depends_on('py-avro@1.8.1:1.10.8', type=('build', 'run'), when='^python@:2.9')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/py-apache-beam/package.py:21]: why is this removed?

Comment (alalazo) [reply var/spack/repos/builtin/packages/py-apache-beam/package.py:21]: Because above we have:
```python
depends_on('python@3:', type=('build', 'run'))
```

**Review by haampie — APPROVED** [review]
